File: mobile application development/front/react-native/src/api/mockApi.ts
import { brokerData } from '../data/repository';
import type {
  AuthResponse,
  LoginRequest,
  PortfolioResponse,
  RegisterRequest,
  TradeRequest,
  TradeResponse,
  UserMe,
} from './types';

const MOCK_JWT = 'mock-jwt';

export function mockRegister(body: RegisterRequest): AuthResponse {
  return {
    token: MOCK_JWT,
    accessToken: MOCK_JWT,
    email: body.email,
    username: body.username,
    name: body.name,
    clntId: 100001,
    roles: ['USER'],
    balance: brokerData.cashRub,
    user: {
      id: 'mock-user',
      userId: 1,
      clntId: 100001,
      email: body.email,
      username: body.username,
      name: body.name,
      roles: ['USER'],
      status: 'ACTIVE',
      balance: brokerData.cashRub,
    },
  };
}

export function mockLogin(body: LoginRequest): AuthResponse {
  return {
    token: MOCK_JWT,
    accessToken: MOCK_JWT,
    email: body.email,
    username: body.email.includes('@') ? body.email.split('@')[0] : body.email,
    name: 'Demo Trader',
    clntId: 100001,
    roles: ['USER'],
    balance: brokerData.cashRub,
    user: {
      id: 'mock-user',
      userId: 1,
      clntId: 100001,
      email: body.email,
      username: body.email.includes('@') ? body.email.split('@')[0] : body.email,
      name: 'Demo Trader',
      roles: ['USER'],
      status: 'ACTIVE',
      balance: brokerData.cashRub,
    },
  };
}

export function mockGetMe(
  email: string,
  username: string | null,
  name: string | null,
  balance: number,
  clntId: number | null
): UserMe {
  return {
    id: 'mock-user',
    userId: 1,
    clntId: clntId ?? 100001,
    email,
    username,
    name,
    roles: ['USER'],
    status: 'ACTIVE',
    balance,
  };
}

export function mockGetPortfolio(): PortfolioResponse {
  return {
    positions: brokerData.portfolioHoldings.map((h) => ({
      instrumentId: h.sym,
      symbol: h.sym,
      quantity: h.qty,
    })),
    cashRub: brokerData.cashRub,
  };
}

export function mockExecuteTrade(body: TradeRequest, side: 'buy' | 'sell'): TradeResponse {
  const tk = brokerData.findTicker(body.instrumentId);
  const stats = brokerData.dayStats(tk);
  return {
    id: `mock-${Date.now()}`,
    instrumentId: body.instrumentId,
    symbol: tk.sym,
    side,
    quantity: body.quantity,
    price: stats.price,
    currency: tk.ccy,
    executedAt: Date.now(),
  };
}

File: mobile application development/front/kotlin/app/src/main/java/com/itmo/mybroker/api/MockApi.kt
package com.itmo.mybroker.api

import com.itmo.mybroker.data.BrokerData

private const val MOCK_JWT = "mock-jwt"

object MockApi {
    fun register(body: RegisterRequest): AuthResponse {
        return AuthResponse(
            token = MOCK_JWT,
            email = body.email,
            username = body.username,
            name = body.name,
            clntId = 1,
            roles = listOf("USER"),
            balance = BrokerData.repository.cashRub,
        )
    }

    fun login(body: LoginRequest): AuthResponse =
        AuthResponse(
            token = MOCK_JWT,
            email = body.email,
            username = "mock_trader",
            name = BrokerData.repository.user.nameRu,
            clntId = 1,
            roles = listOf("USER"),
            balance = BrokerData.repository.cashRub,
        )

    fun getMe(email: String, username: String?, name: String?, balance: Double): UserMe =
        UserMe(
            id = "mock-user",
            userId = 1,
            clntId = 1,
            email = email,
            username = username ?: "mock_trader",
            name = name ?: BrokerData.repository.user.nameRu,
            roles = listOf("USER"),
            status = "ACTIVE",
            balance = balance,
        )

    fun getPortfolio(): PortfolioResponse =
        PortfolioResponse(
            positions = BrokerData.repository.portfolioHoldings.map {
                PortfolioPosition(instrumentId = it.sym, symbol = it.sym, quantity = it.qty.toDouble())
            },
            cashRub = BrokerData.repository.cashRub,
        )

    fun executeTrade(body: TradeRequest, side: String): TradeResponse {
        val tk = BrokerData.repository.bySym(body.instrumentId)
        val stats = BrokerData.repository.dayStats(tk)
        return TradeResponse(
            id = "mock-${System.currentTimeMillis()}",
            instrumentId = body.instrumentId,
            symbol = tk.sym,
            side = side,
            quantity = body.quantity,
            price = stats.price,
            currency = tk.ccy.name,
            executedAt = System.currentTimeMillis(),
        )
    }
}
